src/firefox/index.js

```diff
@@ -17,6 +17,82 @@ import {default as defaultFirefoxConnector, REMOTE_PORT} from './remote';
 
 const log = createLogger(__filename);
 
+// Flow Types
+
+import type {DefaultFirefoxConnectorFn} from './remote';
+import type {ChildProcess} from 'child_process';
+import type {
+  PreferencesAppName,
+  PreferencesGetterFn,
+} from './preferences';
+
+import type {ExtensionManifest} from '../util/manifest';
+
+export type FirefoxRunnerParams = {
+  binary?: string,
+  profile?: string,
+  "new-instance"?: boolean,
+  "no-remote"?: boolean,
+  "foreground"?: boolean,
+  "listen"?: number,
+  "binary-args"?: Array<string> | string,
+  "env"?: {
+    [key: string]: string
+  },
+  "verbose"?: boolean,
+};
+
+export type FirefoxRunnerResults = {
+  process: ChildProcess,
+  binary: string,
+  args: Array<string>,
+};
+
+export type FirefoxRunnerFn =
+  (params: FirefoxRunnerParams) => FirefoxRunnerResults;
```

Comment:
@kumar303 yeah, that was a thing that I definitely wanted to look before fixing it:

basically flow couldn't catch the above mistake in the type signature because, in the only place where we use it, we are returning its return value in the middle of a Promise chain, where
both the signatures would become a "Promise<FirefoxRunnerResults>" in any case (with both the right and the wrong type signature), on the contrary flow would be able to detect the wrong signature if we tried to use the return value directly.